.github/workflows/test-wheel-windows.yml

```diff
@@ -84,11 +84,23 @@ jobs:
         continue-on-error: true
 
       - name: Update driver
+        env:
+          DRIVER_MODE: ${{ matrix.DRIVER_MODE }}
+          GPU_TYPE: ${{ matrix.GPU }}
         run: |
-          .github/workflows/install_gpu_driver.ps1
+          ci/tools/install_gpu_driver.ps1
 
       - name: Ensure GPU is working
-        run: nvidia-smi
+        run: |
+          nvidia-smi
+
+          $mode_output = nvidia-smi | Select-String -Pattern "${{ matrix.DRIVER_MODE }}"
+          Write-Output "Driver mode check: $mode_output"
+          if ("$mode_output" -eq "") {
+            Write-Error "Switching to driver mode ${{ matrix.DRIVER_MODE }} failed!"
+            exit 1
+          }
```

Comment:
Example for why this check is needed: https://github.com/NVIDIA/cuda-python/actions/runs/19416448057/job/55546003709?pr=1242#step:6:37
If mode switching fails, `nvidia-smi` would just print a warning and move on silently. This detection helps us ensure our test matrix is correctly specified.